src/bigquery.ts

```diff
@@ -0,0 +1,1776 @@
+        (callback as SimpleQueryRowsCallback)(null, [], resp);
+        return;
+      }
+      // The Job is important for the `queryAsStream_` method, so a new query
+      // isn't created each time results are polled for.
+      options = extend({job}, options);
+      job!.getQueryResults(options, callback as QueryRowsCallback);
+    });
+  }
+
+  /**
+   * This method will be called by `createQueryStream()`. It is required to
+   * properly set the `autoPaginate` option value.
+   *
+   * @private
+   */
+  queryAsStream_(query: Query, callback?: SimpleQueryRowsCallback) {
+    if (query.job) {
+      query.job.getQueryResults(query, callback as QueryRowsCallback);
+      return;
+    }
+    this.query(query, {autoPaginate: false}, callback);
+  }
+}
+
+/*! Developer Documentation
+ *
+ * These methods can be auto-paginated.
+ */
+paginator.extend(BigQuery, ['getDatasets', 'getJobs']);
+
+/*! Developer Documentation
+ *
+ * All async methods (except for streams) will return a Promise in the event
+ * that a callback is omitted.
+ */
+promisifyAll(BigQuery, {
+  exclude: [
+    'dataset',
+    'date',
+    'datetime',
+    'geography',
+    'job',
+    'time',
+    'timestamp',
+  ],
+});
+
+/**
+ * Date class for BigQuery.
+ */
+export class BigQueryDate {
+  value: string;
+  constructor(value: BigQueryDateOptions | string) {
+    if (typeof value === 'object') {
+      value = BigQuery.datetime(value).value;
+    }
+    this.value = value;
+  }
+}
+
+/**
+ * Geography class for BigQuery.
+ */
+export class Geography {
+  value: string;
+  constructor(value: string) {
+    this.value = value;
+  }
+}
+
+/**
+ * Timestamp class for BigQuery.
+ */
+export class BigQueryTimestamp {
+  value: string;
+  constructor(value: Date | string) {
+    this.value = new Date(value).toJSON();
+  }
+}
+
+/**
+ * Datetime class for BigQuery.
+ */
+export class BigQueryDatetime {
+  value: string;
+  constructor(value: BigQueryDatetimeOptions | string) {
+    if (typeof value === 'object') {
+      let time;
+      if (value.hours) {
+        time = BigQuery.time(value).value;
+      }
+      value = format('{y}-{m}-{d}{time}', {
+        y: value.year,
+        m: value.month,
+        d: value.day,
+        time: time ? ' ' + time : '',
+      });
+    } else {
+      value = value.replace(/^(.*)T(.*)Z$/, '$1 $2');
+    }
+    this.value = value as string;
+  }
+}
+
+/**
+ * Time class for BigQuery.
+ */
+export class BigQueryTime {
+  value: string;
+  constructor(value: BigQueryTimeOptions | string) {
+    if (typeof value === 'object') {
+      value = format('{h}:{m}:{s}{f}', {
+        h: value.hours,
+        m: value.minutes || 0,
+        s: value.seconds || 0,
+        f: is.defined(value.fractional) ? '.' + value.fractional : '',
+      });
+    }
+    this.value = value as string;
+  }
+}
+
+/**
+ * {@link Dataset} class.
+ *
+ * @name BigQuery.Dataset
+ * @see Dataset
+ * @type {constructor}
+ */
+export {Dataset};
```

Comment:
We could probably get rid of all these exports since the index file should be importing/exporting from their actual source files now.